### DISCUSSION TITLE
Why do I only import winston and not use winston, and the size of the budle will increase ?

### DISCUSSION THREAD

**Discussion by zerofront** [vercel/next.js#22410] [Help]
**What version of Next.js are you using?**

latest

**What version of Node.js are you using?**

12

**What browser are you using?**

chrome

**What operating system are you using?**

macOS

**How are you deploying your application?**

next build

**Describe the Bug**

Why do I only import winston and not use winston, and the size of the budle will increase ?

**Expected Behavior**

increase the size of the budle  only when the module is uesd

**To Reproduce**

```
import winston from 'winston';

export default function serverFunction () {
  const IS_SERVER = typeof window === 'undefined';

  if (!IS_SERVER) {
    console.log('error')
  } else {
    console.log('serverFunction')
  }
}
```

```javascript
export default function serverFunction () {
  const IS_SERVER = typeof window === 'undefined';

  if (!IS_SERVER) {
    console.log('error')
  } else {
    console.log('serverFunction')
  }
}
```
![image](https://user-images.githubusercontent.com/19327366/108582903-f4787b00-7370-11eb-8b66-3c0e9a9d2618.png)

```javascript
import winston from 'winston';

export default function serverFunction () {
  const IS_SERVER = typeof window === 'undefined';

  if (!IS_SERVER) {
    console.log('error')
  } else {
    console.log('serverFunction')
  }
}
```
![image](https://user-images.githubusercontent.com/19327366/108582896-ef1b3080-7370-11eb-86a6-8fa977d3a14c.png)

**Comment by merceyz** [comment]
Because `winston` has side-effects so webpack can't remove it

**Reply by zerofront** [reply]
How can we know winston has side-effects?
